File: lib/instance_manager.dart
class InstanceManager {
  final Map<Object, int> _instancesToInstanceIds = {};
  final Map<int, Object> _instanceIdsToInstances = {};

  int _nextInstanceId = 0;

  static final InstanceManager instance = InstanceManager();

  int? tryAddInstance(Object instance) {
    if (_instancesToInstanceIds.containsKey(instance)) {
      return null;
    }
    _nextInstanceId++;
    _instancesToInstanceIds[instance] = _nextInstanceId;
    _instanceIdsToInstances[_nextInstanceId] = instance;
    return _nextInstanceId;
  }

  int? removeInstance(Object instance) {
    int? instanceId = _instancesToInstanceIds[instance];
    if (instanceId != null) {
      _instancesToInstanceIds.remove(instance);
      _instanceIdsToInstances.remove(instanceId);
    }

    return instanceId;
  }

  Object? getInstance(int instanceId) {
    return _instanceIdsToInstances[instanceId];
  }

  int? getInstanceId(Object instance) {
    return _instancesToInstanceIds[instance];
  }
}

File: android/src/main/kotlin/com/example/flutter_wangyan_webview/InstanceManager.kt
package com.example.flutter_wangyan_webview

import androidx.collection.LongSparseArray

object InstanceManager {
    private val instanceIdsToInstances = LongSparseArray<Any>()
    private val instancesToInstanceIds = HashMap<Any, Long>()

    fun addInstance(instance: Any, instanceId: Long) {
        instancesToInstanceIds[instance] = instanceId;
        instanceIdsToInstances.append(instanceId, instance);
    }

    fun removeInstanceWithId(instanceId: Long): Any? {
        val instance = instanceIdsToInstances[instanceId]
        if (instance != null) {
            instanceIdsToInstances.remove(instanceId)
            instancesToInstanceIds.remove(instance)
        }
        return instance
    }

    fun removeInstance(instance: Any?): Long? {
        val instanceId = instancesToInstanceIds[instance]
        if (instanceId != null) {
            instanceIdsToInstances.remove(instanceId)
            instancesToInstanceIds.remove(instance)
        }
        return instanceId
    }

    fun getInstance(instanceId: Long): Any? {
        return instanceIdsToInstances[instanceId]
    }

    fun getInstanceId(instance: Any?): Long? {
        return instancesToInstanceIds[instance]
    }
}